EPAS 15 - Merge Statement (#3804)
EnterpriseDB/docs: product_docs/docs/epas/15/epas_compat_sql/65a_merge.mdx

---
title: "MERGE"
---

## Name

`MERGE` &mdash; conditionally insert, update, or delete rows of a table.

## Synopsis

```sql
MERGE INTO target_table_name [ target_alias ]
USING data_source [ source_alias ] ON ( join_condition )
-- merge_update_clause
WHEN MATCHED THEN 
   UPDATE SET column = value, ... [ WHERE where_condition ]
   [ DELETE [ WHERE where_condition ] ]

-- merge_insert_clause
WHEN NOT MATCHED THEN 
   INSERT [(col_list)] VALUES (val_list) [ WHERE where_condition ] 
```

## Description

> `MERGE` allows you to select rows from one or more sources for update or insertion into a table. You specify the join condition to determine whether to update or Insert into the target table. Optionally you specify conditional UPDATE and INSERT statements using the WHERE clause in the MERGE statement.

Reviewer: ```suggestion
`MERGE` allows you to select rows from one or more sources for update or insertion into a table. You specify the join condition to determine whether to update or insert into the target table. Alternatively, you can specify conditional UPDATE and INSERT statements using the WHERE clause in the MERGE statement.
```

> `MERGE` allows you to select rows from one or more sources for update or insertion into a table. You specify the join condition to determine whether to update or Insert into the target table. Optionally you specify conditional UPDATE and INSERT statements using the WHERE clause in the MERGE statement.

Reviewer: ```suggestion
`MERGE` allows you to select rows from one or more sources for update or insertion into a table. You specify the join condition to determine whether to update or insert into the target table. You specify conditional UPDATE and INSERT statements using the WHERE clause in the MERGE statement.
```

```sql
MERGE INTO target_table_name [ target_alias ]
USING data_source [ source_alias ] ON ( join_condition )
-- merge_update_clause
WHEN MATCHED THEN 
   UPDATE SET column = value, ... [ WHERE where_condition ]
   [ DELETE [ WHERE where_condition ] ]

-- merge_insert_clause
WHEN NOT MATCHED THEN 
   INSERT [(col_list)] VALUES (val_list) [ WHERE where_condition ] 
```

## Description

> `MERGE` allows you to select rows from one or more sources for update or insertion into a table. You specify join condition to determine whether to UPDATE or INSERT into the target table. Optionally you specify conditional UPDATE and INSERT using the WHERE clause in the MERGE statement.

Reviewer: Use lowercase when using SQL statement names as verbs instead of nouns. (you might want to check with Betsy about this)
```suggestion
`MERGE` allows you to select rows from one or more sources for update or insertion into a table. You specify the JOIN condition to determine whether to update or insert into the target table. Optionally, you specify conditional UPDATE and INSERT statements using the WHERE clause in the MERGE statement.
```

Author: Applied changes in separate commit

Add rows to both the tables:

```sql
# Insert rows into target table
INSERT INTO target VALUES (1, 0);
INSERT INTO target VALUES (2, 20);
INSERT INTO target VALUES (3, 0);

# Insert rows into source table
INSERT INTO source VALUES (1, 100);
INSERT INTO source VALUES (2, 200);
INSERT INTO source VALUES (3, 300);
INSERT INTO source VALUES (4, 100);
INSERT INTO source VALUES (5, 300);
INSERT INTO source VALUES (6, 600);
```

> This example shows how to use the WHERE condition for conditional UPDATE and INSERT in the MERGE statement:

Reviewer: ```suggestion
This example shows how to use the WHERE condition for conditional UPDATE and INSERT statements in the MERGE statement:
```

Author: Applied changes in separate commit
